pkg/build/controller/scheduler/scheduler.go

```diff
@@ -0,0 +1,143 @@
+package scheduler
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api/errors"
+
+	kerrors "k8s.io/kubernetes/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/util/wait"
+
+	"github.com/golang/glog"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildclient "github.com/openshift/origin/pkg/build/client"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+	"github.com/openshift/origin/pkg/client"
+)
+
+type Scheduler interface {
+	Next(*buildapi.Build) (bool, error)
+}
+
+type ParallelScheduler struct{}
+
+func (s *ParallelScheduler) Next(build *buildapi.Build) (bool, error) {
+	if buildutil.BuildSchedulePolicy(build) != buildapi.BuildSchedulePolicyParallel {
+		return false, nil
+	}
+	glog.V(5).Infof("Build %s/%s scheduler is parallel", build.Namespace, build.Name)
+	return true, nil
+}
+
+type SerialScheduler struct {
+	OSClient client.Interface
+}
+
+func (s *SerialScheduler) Next(build *buildapi.Build) (bool, error) {
+	if buildutil.BuildSchedulePolicy(build) != buildapi.BuildSchedulePolicySerial {
+		return false, nil
+	}
+	glog.V(5).Infof("Build %s/%s scheduler is serial", build.Namespace, build.Name)
+	nextBuild, err := GetNextBuild(s.OSClient.Builds(build.Namespace), build)
+	return (nextBuild != nil && nextBuild.Name == build.Name), err
+}
+
+type LatestOnlyScheduler struct {
+	BuildUpdater buildclient.BuildUpdater
+	OSClient     client.Interface
+}
+
+func (s *LatestOnlyScheduler) Next(build *buildapi.Build) (bool, error) {
+	if buildutil.BuildSchedulePolicy(build) != buildapi.BuildSchedulePolicySerialLatestOnly {
+		return false, nil
+	}
+	glog.V(5).Infof("Build %s/%s scheduler is latest-only", build.Namespace, build.Name)
+	nextBuild, err := GetNextBuild(s.OSClient.Builds(build.Namespace), build)
```

Comment:
not clear to me how this works.  Isn't GetNextBuild still going to return the next build in the list, not the most recent build in the list?